### ISSUE TITLE
Add visualizations to project READMEs as a standard.

### ISSUE THREAD

**Issue by howardroark** [regexhq/regexhq#13]
I have been using the site https://www.debuggex.com/ lately and I love it.  It really helps me digest the nature of a regex.  It would be really neat if each project had a demo of the current state of the regex in the README as a standard.

For example dirname regex...

![](https://www.debuggex.com/i/g2dGhztutRYshDqZ.png)

Or email ...

![](https://www.debuggex.com/i/WyGJ1iuDvCo0BMF_.png)

...

**Comment by tunnckoCore** [comment]
Good idea :+1: 
thanks for the tool, added to wiki

**Comment by sindresorhus** [comment]
:+1: They actually have an embed feature:

![Regular expression visualization](https://www.debuggex.com/i/aB2JISKavfX3PjuG.png)

**Comment by howardroark** [comment]
Yeah, it's so great!  I emailed the project maintainer about offering a service where you could supply your regex as part of the URL.

That way you could do **something like this** this in Markdown...

```
![](https://www.debuggex.com?regex=^([\w_\.\-\+])+\@([\w\-]+\.)+([\w]{2,10})+$)
```

They said that it had been added to their ticketing system.  Though in these cases it may also come with a watermark to identify their product.  Could be super handy for automation though.  With GitHub it could be great for issue conversations.

**Comment by tunnckoCore** [comment]
Yeah! The idea sounds great! Maybe also some more flexible API to be possible CLI automation too xD

**Comment by jonschlinkert** [comment]
yeah that's a great tool. nice idea

**Comment by akileez** [comment]
there is another tool as well… http://jex.im/regulex/  ...and its on github

**Comment by howardroark** [comment]
ooooo, nice!  Theoretically you could make a service that took a regex get param and passed it to a process which took a snapshot of this code with phantomjs and returned it back.  Then you could use it in markdown images.  GitHub also caches all images so the load would be small.

**Comment by tunnckoCore** [comment]
Hmm... maybe with `pageres` with selector capture `#graphCt` which is the div that holds the svg.
**CLI**

``` bash
pageres http://jex.im/regulex/#!embed=false&flags=&re=([^\\\/]*?[^\\\/]*?)((\.([^\\\/.]*))*)$ --selector '#graphCt'
```

or **API**

``` js
var Pageres = require('pageres');

var pageres = new Pageres({delay: 2})
    .src('http://jex.im/regulex/#!embed=false&flags=&re=([^\\\/]*?[^\\\/]*?)((\.([^\\\/.]*))*)$',
{selector: '#graphCt'})
    .dest(__dirname);

pageres.run(function (err) {
    if (err) {
        throw err;
    }

    console.log('done');
});
```

Someone to try it ;p

**Comment by sindresorhus** [comment]
@tunnckoCore Awesome idea! You need to supply Pageres with a resolution though. Works fine then :)

**Comment by sindresorhus** [comment]
I prefer the output of Debuggex though. For now we can easily just create bins there for our repos and use the embed feature.

**Comment by tunnckoCore** [comment]
Yea I realize after that ;d
I prefer to create something like above for now, because debuggex dont have API for now.

We all love automation, huh :D

**Comment by sindresorhus** [comment]
Nah, you'd use more time automating it than it would take to just create those debuggex bins manually. And it doesn't have as much benefit as you would need to manually update the regex in the readme anyways.
